### PR TITLE
Add comprehensive tests for emoji-utils and fix duplicate emoji bug

### PR DESCRIPTION
**Pull request by narulaskaran**
## Summary
- Added 29 comprehensive unit tests for `emoji-utils.ts` (previously untested)
- Fixed duplicate star emoji (⭐) bug in `GROUP_EMOJIS` array
- Achieved 100% test coverage for all emoji utility functions

### Tests Added
- `getAllGroupEmojis()`: 5 tests covering array properties, uniqueness, and validity
- `getRandomGroupEmoji()`: 4 tests covering basic functionality and randomness
- `getRandomGroupEmojiExcluding()`: 7 tests covering exclusion logic and edge cases
- `getUniqueGroupEmoji()`: 10 tests covering unique selection, exhaustion scenarios, and progressive exclusion
- Integration tests: 3 tests for realistic usage scenarios

### Bug Fixed
- Removed duplicate ⭐ emoji from `GROUP_EMOJIS` array
- Array now contains 106 unique emojis (previously 107 with 1 duplicate)

## Test Coverage
All 29 tests passing ✅
- Edge cases: empty arrays, undefined values, exhausted emoji pools
- Probabilistic testing: validates randomness over multiple calls
- Integration testing: realistic group creation scenarios

## Test Plan
- [x] Run `npm test -- emoji-utils.test.ts` - all tests pass
- [x] Verified bug fix resolves duplicate emoji issue
- [x] Confirmed backward compatibility with existing code

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/lib/emoji-utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+} from './emoji-utils';
+
+describe('getAllGroupEmojis', () => {
+  it('returns an array of emojis', () => {
+    const emojis = getAllGroupEmojis();
+    expect(Array.isArray(emojis)).toBe(true);
+    expect(emojis.length).toBeGreaterThan(0);
+  });
+
+  it('returns a copy of the array, not the original', () => {
```
Comment (narulaskaran) [inline src/lib/emoji-utils.test.ts]: Don't care about this, remove

File: src/lib/emoji-utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+    // Call it multiple times to ensure it never returns the excluded emoji
+    for (let i = 0; i < 20; i++) {
+      const emoji = getRandomGroupEmojiExcluding(excluded);
+      expect(emoji).not.toBe(excluded);
+    }
+  });
+
+  it('handles undefined exclusion by returning any emoji', () => {
+    const emoji = getRandomGroupEmojiExcluding(undefined);
```
Comment (narulaskaran) [inline src/lib/emoji-utils.test.ts]: Should not allow undefined vals. Typechecking should address this

File: src/lib/emoji-utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    const emoji = getRandomGroupEmojiExcluding('');
+    expect(typeof emoji).toBe('string');
+    expect(emoji.length).toBeGreaterThan(0);
+  });
+
+  it('handles exclusion of emoji not in the list', () => {
+    const emoji = getRandomGroupEmojiExcluding('🦖'); // Not in the list
+    const allEmojis = getAllGroupEmojis();
+    expect(allEmojis).toContain(emoji);
+  });
```
Comment (narulaskaran) [inline src/lib/emoji-utils.test.ts]: Same thing as the empty string test above

File: src/lib/emoji-utils.test.ts [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+  it('returns different emojis over multiple calls (probabilistic)', () => {
+    const emojis = new Set<string>();
+
+    for (let i = 0; i < 50; i++) {
+      emojis.add(getRandomGroupEmojiExcluding('🐶'));
+    }
+
+    expect(emojis.size).toBeGreaterThan(5);
+    expect(emojis.has('🐶')).toBe(false);
+  });
```
Comment (narulaskaran) [inline src/lib/emoji-utils.test.ts]: Flaky. All tests should be deterministic

**Review by narulaskaran — COMMENTED** [review]
@claude Many of the tests are repetitive or duplicative. Many are not needed. Address  comments and also generally clean up the tests so that they're more concise. We don't need strictly 100% coverage
